SoftLayer/CLI/hardware/credentials.py

```diff
@@ -22,6 +23,12 @@ def cli(env, identifier):
     instance = manager.get_hardware(hardware_id)
 
     table = formatting.Table(['username', 'password'])
+    if 'passwords' not in instance['operatingSystem']:
+        raise exceptions.SoftLayerError("No passwords found in operatingSystem")
+
     for item in instance['operatingSystem']['passwords']:
-        table.add_row([item['username'], item['password']])
```

Comment:
instead of raising an exception, I would prefer this to just print out  "None"

    table.add_row([item.get('username', 'None'), item.get('password', 'None')])

.get() allows you to specify a default value to return if the item isn't found. This article explain it in a bit more detail if you are interested. 
https://docs.quantifiedcode.com/python-anti-patterns/correctness/not_using_get_to_return_a_default_value_from_a_dictionary.html

Reply:
Done